### PR TITLE
Add 'TYPE' argument to choose between 'STATIC' and 'OBJECT' libraries

### LINKED ISSUE #28: Generate `OBJECT` library instead of `STATIC`
When installing a `STATIC` library that includes files using `cmrc_add_resource_library`, it is necessary to export and install the library generated by `CMakeRC` as well. This shouldn't be necessary.
An option would be to generate an `OBJECT` library instead of a `STATIC` library, and to use it with `target_sources` and `target_include_directories` instead of linking it with `target_link_libraries`

### PR DESCRIPTION
**Pull request by drdanz**
Fixes #28

### CODE REVIEW

File: CMakeRC.cmake [hunk 1]
```diff
@@ -410,6 +410,15 @@ function(cmrc_add_resource_library name)
...
     set(libname "${name}")
+    # Check that type is either "STATIC" or "OBJECT", or default to "STATIC" if
+    # not set
+    if(NOT DEFINED ARG_TYPE)
+        set(ARG_TYPE STATIC)
+    elseif(NOT "${ARG_TYPE}" MATCHES "^(STATIC|OBJECT)$")
+        message(SEND_ERROR "${ARG_TYPE} is not a valid TYPE (STATIC and OBJECT are acceptable)")
+        set(ARG_TYPE STATIC)
+    endif()
+    message("ARG_TYPE = ${ARG_TYPE}")
```
Comment (vector-of-bool) [inline CMakeRC.cmake:421]: Have a debug line left over.

**Review by vector-of-bool — CHANGES_REQUESTED** [review]
Thanks! One minor change requested. Otherwise, LGTM.

File: CMakeRC.cmake (hunk 1 above)
Comment (drdanz) [reply CMakeRC.cmake]: Fixed.

**Comment by drdanz** [comment]
Should be fixed now...

**Review by vector-of-bool — APPROVED** [review]

**Pull request merged**